src/ddmd/statementsem.d

```diff
@@ -483,6 +490,366 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
+                        setError();
+                        mixin(returnEarly);
+                    }
+                }
+                Initializer ie = new ExpInitializer(Loc(), new IntegerExp(k));
+                auto var = new VarDeclaration(loc, p.type, p.ident, ie);
+                var.storage_class |= STCmanifest;
+                static if(isStatic) var.storage_class |= STClocal;
+                static if(!isDecl)
+                {
+                    st.push(new ExpStatement(loc, var));
+                }
+                else
+                {
+                    st.push(var);
+                }
+                p = (*fs.parameters)[1]; // value
+            }
+            bool declareVariable(StorageClass storageClass, Type type, Identifier ident, Expression e, Type t)
+            {
+                enum returnEarly = q{ return false; };
+                if (storageClass & (STCout | STClazy) ||
+                    storageClass & STCref && !te)
+                {
+                    fs.error("no storage class for value `%s`", ident.toChars());
+                    setError();
+                    mixin(returnEarly);
+                }
+                Declaration var;
+                if (e)
+                {
+                    Type tb = e.type.toBasetype();
+                    Dsymbol ds = null;
+                    if (!(storageClass & STCmanifest))
+                    {
+                        if ((isStatic || tb.ty == Tfunction || tb.ty == Tsarray || storageClass&STCalias) && e.op == TOKvar)
+                            ds = (cast(VarExp)e).var;
+                        else if (e.op == TOKtemplate)
+                            ds = (cast(TemplateExp)e).td;
+                        else if (e.op == TOKscope)
+                            ds = (cast(ScopeExp)e).sds;
+                        else if (e.op == TOKfunction)
+                        {
+                            auto fe = cast(FuncExp)e;
+                            ds = fe.td ? cast(Dsymbol)fe.td : fe.fd;
+                        }
+                    }
+                    else if (storageClass & STCalias)
+                    {
+                        fs.error("foreach loop variable cannot be both enum and alias");
+                        setError();
+                        mixin(returnEarly);
+                    }
+
+                    if (ds)
+                    {
+                        var = new AliasDeclaration(loc, ident, ds);
+                        if (storageClass & STCref)
+                        {
+                            fs.error("symbol `%s` cannot be ref", ds.toChars());
+                            setError();
+                            mixin(returnEarly);
+                        }
+                        if (paramtype)
+                        {
+                            fs.error("cannot specify element type for symbol `%s`", ds.toChars());
```

Comment:
Why is the variable named parameter type while the error message says element type?